pennylane/transforms/core/transform_program.py

```diff
@@ -510,17 +511,13 @@ def _set_all_argnums(self, qnode, args, kwargs, argnums):
             argnums = [0] if qnode.interface in ["jax", "jax-jit"] and argnums is None else argnums
             # pylint: disable=protected-access
             if (transform._use_argnum or transform.classical_cotransform) and argnums:
-                params = _jax_argnums_to_tape_trainable(
-                    qnode, argnums, TransformProgram(self[0:index]), args, kwargs
-                )
+                params = _jax_argnums_to_tape_trainable(qnode, argnums, self[:index], args, kwargs)
                 argnums_list.append([qml.math.get_trainable_indices(param) for param in params])
             else:
                 argnums_list.append(None)
 
         self._argnums = argnums_list
 
-        qnode.construct(args, kwargs)
```

Comment:
Was this simply unnecessary?

Reply:
Was necessary before recent change to use `make_tape` instead of `construct + qnode.tape`.  Further motivator for why using `make_tape` is a better option here.